### ISSUE TITLE
Bug: Resource List items with a checkbox (selectable)

### ISSUE THREAD

**Issue by resistorsoftware** [baoagency/polaris_view_components#345]
Render an item in a list with this code:

```
           <%= polaris_resource_item(selectable: true) do |resource_item| %>
                  <% resource_item.with_checkbox(
                    name: "reservation[]",
                    value: reservation["id"]
                  ) %>
```

Examine the DOM. Notice that each checkbox input element rendered has an id of **reservation_**. The parent **span** element takes on an ID if one is provided to this component, not the input element child.

I understand this is an example coming from the Lookbook, but it could trip people up. Best we render a unique ID for each input element as some kind of default to ensure any code actually interacting with selected elements works too.

**Comment by resistorsoftware** [comment]
Even stranger and exposing my ignorance perhaps of how this Stimulus works, the rendering of that PVC produces some DOM element that do make some sense:

```
<span data-event-target="span" class="Polaris-Checkbox Polaris-Checkbox--labelHidden">
  <input type="checkbox" name="reservation[]" id="reservation_" value="1493-13" aria-checked="false" class="Polaris-Checkbox__Input">
  <span class="Polaris-Checkbox__Backdrop"></span>
  <span class="Polaris-Checkbox__Icon">
    <span class="Polaris-Icon">
      <svg viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg" class="Polaris-Icon__Svg" focusable="false" aria-hidden="true"><path fill-rule="evenodd" d="M14.03 7.22a.75.75 0 0 1 0 1.06l-4.5 4.5a.75.75 0 0 1-1.06 0l-2.25-2.25a.75.75 0 1 1 1.06-1.06l1.72 1.72 3.97-3.97a.75.75 0 0 1 1.06 0Z"></path></svg>
    </span>
  </span>
</span>
```

Now I have a target in my event controller, set to "span", for the resource item, thinking I can get those spans, and then just get the child element, the input and have the value I covet. Ha! You know what I get in the Stimulus targets? The span below the input element. That's right. For each checkbox I get instead:
` <span class="Polaris-Checkbox__Backdrop"></span>`

So I guess the input I am interested in is this elements Previous Sibling. 

Honestly. This exercise is frustrating! Seems like while these components render pretty, the reality of using them is far from pretty. Obviously someone smart, not me, is giggling at how dumb I am. I care not. It is here for all to see.

**Comment by resistorsoftware** [comment]
I changed my target from a **span** to instead a rando target name, and now I get the wrapping span, and one of the children nodes is the element I am interested in. Wow. Such long-winded computing to get a damn checkbox value. OK. Moving on...

**Comment by resistorsoftware** [comment]
This is truly stupid. You can select as many element as you want, and none of them reflect the fact they are selected. I officially think PVC for this purpose is beyond my ability to use, and I will die on that hill till someone shows me the light.

**Comment by resistorsoftware** [comment]
I started trying to fix this. In the Polaris ResourceItem code I added in an event listener on clicks. Yay! The result of a single click on a resource item checkbox registers two events. The first is a who cares. The second is interesting. Again, the ID being bad will bite here. 

```
click { target: span.Polaris-Checkbox__Backdrop, buttons: 0, clientX: 639, clientY: 628, layerX: 12, layerY: 10 }
click { target: input#reservation_.Polaris-Checkbox__Input, buttons: 0, clientX: 639, clientY: 628, layerX: 12, layerY: 10 }
```
So with that click coming in the door, I can get at the element and hopefully add the "checked" attribute that would make this actually a useful PVC component, a selectable resource item.
